conan/cli/commands/workspace.py

```diff
@@ -237,6 +234,71 @@ def workspace_init(conan_api: ConanAPI, parser, subparser, *args):
+    # Basic collaborators: remotes, lockfile, profiles
+    remotes = conan_api.remotes.list(args.remote) if not args.no_remote else []
+    overrides = eval(args.lockfile_overrides) if args.lockfile_overrides else None
+    # The lockfile by default if not defined will be read from the root workspace folder
+    ws_folder = conan_api.workspace.folder()
+    lockfile = conan_api.lockfile.get_lockfile(lockfile=args.lockfile, conanfile_path=ws_folder,
+                                               cwd=None,
+                                               partial=args.lockfile_partial, overrides=overrides)
+    profile_host, profile_build = conan_api.profiles.get_profiles_from_args(args)
+    print_profiles(profile_host, profile_build)
+
+    ConanOutput().box("Exporting workspace packages recipes to Conan cache")
+    exported_refs = conan_api.workspace.export()
+
+    build_mode = args.build if args.build else []
+    build_mode.extend(f"missing:{r}" for r in exported_refs)
+
+    all_packages = conan_api.workspace.editable_packages
+    packages = conan_api.workspace.select_packages(args.pkg)
+    conan_api.workspace.enable(False)
+
+    install_order = conan_api.workspace.build_order(packages, profile_host, profile_build, build_mode,
+                                                    lockfile, remotes, args, update=args.update)
+
+    ConanOutput().box("Workspace creating each package")
+    order = install_order.install_build_order()
+
+    profile_args = ProfileArgs.from_args(args)
+    for level in order["order"]:
+        for elem in level:
+            ref = RecipeReference.loads(elem["ref"])
+            for package_level in elem["packages"]:
+                for package in package_level:
+                    if package["binary"] != "Build":
+                        continue
+
+                    if ref not in all_packages:
+                        # Build external to Workspace
+                        cmd = f'install {package["build_args"]} {profile_args}'
+                        ConanOutput().box(f"Workspace building external {ref}")
+                        ConanOutput().info(f"Build command: {cmd}\n")
+                        conan_api.command.run(cmd)
+                    else:  # Package in workspace
+                        path = packages[ref]["path"]
+                        # TODO: Missing --lockfile-overrides arg here
+                        build = "--build-require" if package["context"] == "build" else ""
+                        ref_args = " ".join(f"--{k}={getattr(ref, k)}"
+                                            for k in ("name", "version", "user", "channel")
+                                            if getattr(ref, k, None))
+                        cmd = f'create "{path}" {profile_args} {build} {ref_args}'
+                        ConanOutput().box(f"Workspace create {ref}")
+                        ConanOutput().info(f"Conan create command: {cmd}\n")
+                        conan_api.command.run(cmd)
```

Comment:
I reslly like this idea of directly calling the command with the args instead of redoing the command logic here 👍